src/sentry/models/organizationmember.py

```diff
@@ -472,19 +457,43 @@ def get_allowed_roles_to_invite(self):
         Return a list of roles which that member could invite
         Must check if member member has member:admin first before checking
         """
-        return [r for r in roles.get_all() if r.priority <= roles.get(self.role).priority]
+        return [
+            r
+            for r in organization_roles.get_all()
+            if r.priority <= organization_roles.get(self.role).priority
+        ]
 
     def is_only_owner(self) -> bool:
-        if self.role != roles.get_top_dog().id:
+        if self.role != organization_roles.get_top_dog().id:
             return False
 
         return (
             not OrganizationMember.objects.filter(
                 organization=self.organization_id,
-                role=roles.get_top_dog().id,
+                role=organization_roles.get_top_dog().id,
                 user__isnull=False,
                 user__is_active=True,
             )
             .exclude(id=self.id)
             .exists()
         )
+
+    def update_role(self, role: str) -> None:
```

Comment:
That's fine. My (admittedly not Pythonic) instinct is to have the role attribute be effectively private, and to have a model method enforce the logic for changing it. But there's no way to enforce that other than convention and I doubt we'll be altering the attribute anywhere else, so moving the code to where it's called should be fine.

Reply:
yeah fwiw i would just prefix them with `_` if you want to communicate that expectation. doesnt need to be an instance method at that piont - anywhere i would see `def _foo()` I would assume its private to local space only and likely a hairy API that you shouldn't reuse without understanding

Reply:
Is it normal for us to do that with model fields, though? That is,
```python
    _role = models.CharField(max_length=32, null=True, blank=True)
```
Because my wish would be to prevent those who don't know better from directly modifying the field as in `member.role = 'admin'` or `member.update(role='admin')` and instead to guide them toward `member.update_role('admin')` as the public method.

As far as I can find, we don't use underscore-prefixed field names in any models (and it would propagate into SQL if we did, looking odd in Redash, etc.).

Reply:
ah sorry i thought you meant the helper methods

you can technically do that.. id say its not typical and im sure theres lots of strong opinions. like ive definitely setup model fields that are "private" and have getter/setter behavior to ensure its normalized

i think the 'more correct' (albiet more code) way to do that these days is to create a custom django model field that does the validation/serialization/deserialization for you

Reply:
> i think the 'more correct' (albiet more code) way to do that these days is to create a custom django model field that does the validation/serialization/deserialization for you

Hmm, thanks for the tip. I'll look into that.